pinot-controller/src/main/java/com/linkedin/pinot/controller/ControllerConf.java

```diff
@@ -89,6 +102,24 @@ public static String constructDownloadUrl(String tableName, String segmentName,
     }
   }
 
+  public static String constructDownloadUrlWithVersion(String tableName, String segmentName, String vip, SegmentVersion segmentVersion) {
+    try {
+      return StringUtil.join("/", vip, "segments", tableName, URLEncoder.encode(segmentName, "UTF-8") + "?version=" + segmentVersion.toString());
+    } catch (UnsupportedEncodingException e) {
+      // Shouldn't happen
+      throw new AssertionError("UTF-8 encoding should always be supported", e);
+    }
+  }
+
+  public void setPinotFSFactoryClass(Configuration pinotFSFactoryClasses) {
+    setProperty(PINOT_FS_FACTORY_CLASS_LOCAL, DEFAULT_PINOT_FS_FACTORY_CLASS_LOCAL);
+    setProperty(PINOT_FS_FACTORY_CLASS_HADOOP, DEFAULT_PINOT_FS_FACTORY_CLASS_HADOOP);
```

Comment:
Its not clear why we are doing this.

Reply:
I want the default open source classes to be included in the config so users don't wire those in themselves.